### ISSUE TITLE
Change Project slug to include owner name?

### ISSUE THREAD

**Issue by edpaget** [zooniverse/Panoptes#1139]
Any thoughts on making the project slug include the owner-name by default when the response is serialized? eg

```json
{
  "projects": {
    ...,
    "slug": "edwardothegreat/edwardos-sad-project"
  }
}
```

And similarly changing the slug filter param to reflect this `GET /api/projects?slug=edwardothegreat%2Fedwardos-sad-project`.

I think this would definitely benefit the front-end implementation in terms of saving extra requests/includes, but it would also make the `slug` make more sense since it will fully uniquely identify a project by itself.

**Comment by camallen** [comment]
Sounds useful. I can't think of any drawbacks at the moment. Are there any you can outline?

**Comment by aweiksnar** [comment]
:+1: I'm in favor of this

**Comment by edpaget** [comment]
There's the database hit, but that could be covered up by eagerloading, and we incur it anyway since almost all the time people are doing `GET /api/projects?include=owner`.

**Comment by parrish** [comment]
> I think this would definitely benefit the front-end implementation in terms of saving extra requests/includes

`api.type('projects').get({ slug: 'edwardos-sad-project', owner: 'edwardothegreat', include: ['background', 'pages', 'workflows'] })` seems to work just fine, so I'm not sure why that would reduce the number of calls.  (Apparently `owner` isn't a valid include)

>  it would also make the slug make more sense since it will fully uniquely identify a project by itself

Generally agreed

**Comment by parrish** [comment]
Yep, just double checked that: `{"errors":[{"message":":owner is not a valid include for Project"}]}`

**Comment by edpaget** [comment]
Sorry I mistyped it 'owners'.

**Comment by parrish** [comment]
Yep, just caught that myself

**Comment by edpaget** [comment]
It would reduce the number of calls if you want to build a link to the Project's page, since to do the `api.type('projects').get({ slug: 'edwardos-sad-project', owner: 'edwardothegreat', include: ['background', 'pages', 'workflows'] }` you have to know the owner name which is an additional request at the moment.

**Comment by parrish** [comment]
Not in general,  it's a URL param.  #/projects/edwardothegreat/edwardos-sad-project

**Comment by edpaget** [comment]
Only one case though. I ran into this building links from Collection subjects to their Talk pages. Also the /projects route.

**Comment by chrissnyder** [comment]
+1, if for no other reason than to simplify link generation on the FE. Tons of cases in Talk where this would make life a bit easier.

**Comment by parrish** [comment]
Ah, the `#/projects` route is building the urls with the owner name (which could reduce an include if the slug was the formed url): `projects?include=owners,avatar&launch_approved=true`

The collection would have to make an additional call since the links only include the project id.

If we ubiquitously want projects linked and slugged like that, shouldn't we also setup the api response to link to the project via the slug?

**Comment by edpaget** [comment]
> If we ubiquitously want projects linked and slugged like that, shouldn't we also setup the api response to link to the project via the slug?

What do you mean?

**Comment by parrish** [comment]
Links specify an href to use.  Why not specify the slugged href in the link?  That would cut out all requests/includes to fetch the slug

**Comment by edpaget** [comment]
The lab doesn't link to projects that way unfortunately, although there's no reason it couldn't I think I'll start by changing the slug field and then we can look into other changes.

**Comment by brian-c** [comment]
:+1:, although I'm not sure how the react-router'll handle a param containing a slash.

**Comment by camallen** [comment]
was this closed by #1219 ?

**Comment by edpaget** [comment]
Si
